solidity/contracts/factory.sol

```diff
@@ -20,54 +20,67 @@ import {IZetoFungibleInitializable} from "./lib/interfaces/zeto_fungible_initial
 import {IZetoNonFungibleInitializable} from "./lib/interfaces/zeto_nf_initializable.sol";
 
 contract ZetoTokenFactory {
+    struct InitArgs {
+        address implementation;
+        address depositVerifier;
+        address withdrawVerifier;
+        address verifier;
+    }
+
     event ZetoTokenDeployed(address indexed zetoToken);
 
-    mapping(string => address) internal implementations;
+    mapping(string => InitArgs) internal implementations;
 
     constructor() {}
 
     function registerImplementation(
         string memory name,
```

Comment:
what is this name? wouldn't it just be the address?

Reply:
this allows the factory to clone/deploy a new implementation by being passed just a name for the token implementation such as `Zeto_AnonNullifier`